### PR TITLE
render preview via sw

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

**Comment by sebastianbenz** [comment]
@kul3r4 PTAL

**Comment by sebastianbenz** [comment]
Deployed here: https://amp-by-example-staging.appspot.com/playground/#url=https%3A%2F%2Famp-by-example-staging.appspot.com%2Fcomponents%2Famp-form%2Fsource%2F

**Comment by sebastianbenz** [comment]
@dandv PTAL
@amedina FYI

File: playground/src/elements/webui-preview.html [hunk 1]
```diff
@@ -37,17 +37,35 @@
...
+            // store preview doc in the cache to return it via the serviceworker
+            caches.open('previews').then(function(cache) {
+              const data = { headers: {'Content-Type': 'text/html'} };
+              cache.put(
+                  new Request(PREVIEW_REQUEST_URL), 
+                  new Response(documentString, data));
+              previewIframe.src = PREVIEW_REQUEST_URL;
+            });
+          } else {
+            console.log('[preview] No serviceworker detected. Falling back to ' +
```
Comment (kul3r4) [inline playground/src/elements/webui-preview.html]: If no service worker is detected, samples will not work when offline, right? Should we add "offline" to the sentence?

Comment (sebastianbenz) [inline playground/src/elements/webui-preview.html]: Samples will work when offline (if no sw is available no network request will be made). The only thing not working without sw is samples requiring cors.

Comment (kul3r4) [inline playground/src/elements/webui-preview.html]: right, thanks for explaining!
LGTM

**Pull request merged**